src/lib.rs

```diff
@@ -636,7 +636,7 @@ pub struct DirEntry {
     /// We use this to determine whether an entry is a directory or not, which
     /// works around a bug in Rust's standard library:
     /// https://github.com/rust-lang/rust/issues/46484
-    #[cfg(windows)]
+    #[cfg(not(unix))]
```

Comment:
Hmm, I'm not sure I understand this. From inspection, this field is only ever accessed behind a `cfg(windows)` gate, so it seems like this should remain `cfg(windows)`. What exactly does this fix?

Looking more closely at things, it actually seems like you probably want to delete this line, since neither `ino` nor `metadata` are present for `cfg(not(any(unix, windows)))`: https://github.com/BurntSushi/walkdir/blob/e1f7637cf83fe0c8ba9871595ccede4b045640ce/src/lib.rs#L1200

Reply:
I need to check that the latest walkdir works - it looks like the change I made is out of date.